Source/carthage/Archive.swift

```diff
@@ -16,19 +16,21 @@ public struct ArchiveCommand: CommandType {
 	public struct Options: OptionsType {
 		public let outputPath: String
 		public let directoryPath: String
+		public let derivedDataPath: String
 		public let colorOptions: ColorOptions
 		public let frameworkNames: [String]
 
-		static func create(outputPath: String) -> String -> ColorOptions -> [String] -> Options {
-			return { directoryPath in { colorOptions in { frameworkNames in
-				return self.init(outputPath: outputPath, directoryPath: directoryPath, colorOptions: colorOptions, frameworkNames: frameworkNames)
-			} } }
+		static func create(outputPath: String) -> String -> String -> ColorOptions -> [String] -> Options {
+			return { directoryPath in { derivedDataPath in { colorOptions in { frameworkNames in
+				return self.init(outputPath: outputPath, directoryPath: directoryPath, derivedDataPath: derivedDataPath, colorOptions: colorOptions, frameworkNames: frameworkNames)
+			} } } }
 		}
 
 		public static func evaluate(m: CommandMode) -> Result<Options, CommandantError<CarthageError>> {
 			return create
 				<*> m <| Option(key: "output", defaultValue: "", usage: "the path at which to create the zip file (or blank to infer it from the first one of the framework names)")
 				<*> m <| Option(key: "project-directory", defaultValue: NSFileManager.defaultManager().currentDirectoryPath, usage: "the directory containing the Carthage project")
+				<*> m <| Option(key: "derived-data", defaultValue: "", usage: "path to the custom derived data folder")
```

Comment:
It doesn't seem like this should be an argument to `carthage archive`. I don't see how it would affect the behavior.